src/modules/collection/sagas.ts

```diff
@@ -379,12 +379,10 @@ function* handleRequestCollectionSuccess() {
   try {
     const collections: Collection[] = yield select(getWalletCollections)
 
+    // TODO: This is done synchronously, is this something we can / want to change?
```

Comment:
Changing this might not be easy, but do we want it to work synchronously for some reason?

Reply:
Maybe we can create a ticket and refactor this with some testing involved.